tests/src/tests.rs

```diff
@@ -567,17 +570,22 @@ fn test_part(
     // however, as the line of the hint is still verified.
     let mut actual_diagnostics = HashSet::new();
     for diagnostic in &diagnostics {
-        // Ignore diagnostics from other files.
-        if diagnostic.span.id().map_or(false, |id| id != source.id()) {
```

Comment:
@\reviewer this may be controversial, so see an alternative I tried here:

https://github.com/typst/typst/pull/1872/commits/74f8298a584002ca4864776e22d8f55724c600da#diff-5f52c999917aa129d575cd2039a3b67aa67374f8e8328474a9e8f8ef89286e0cR367-R578

It is possible to add a flag, which would allow current tests to stay as-is.

I prefer the new variant, as it actually verifies certain behaviours as errors instead of just having a comment describing them but the diagnostics never being used for the test validation. See the adjusted .typ files (especially with cyclical dependencies)